pil-analyzer/tests/types.rs

```diff
@@ -768,32 +768,68 @@ fn wrong_struct() {
     let input = "
     struct Dot { x: int, y: int }
     let f: int -> Dot = |i| NotADot{x: 0, y: i};
-    let x = f(0);
     ";
     type_check(input, &[]);
 }
 
 #[test]
-#[should_panic = "Field 'a' not found in struct 'Dot'"]
+#[should_panic = "Struct 'Dot' has no field named 'a'"]
 fn struct_wrong_fields() {
     let input = "
     struct Dot { x: int, y: int }
-    let f: int -> Dot = |i| Dot{a: 0, y: i};
+    let f: int -> Dot = |i| Dot{x: 0, y: i, a: 2};
     let x = f(0);
     ";
     type_check(input, &[]);
 }
 
 #[test]
-#[should_panic = "Field 'z' is declared but never used in struct 'Point' initialization"]
+#[should_panic = "Missing field 'z' in initializer of 'A'"]
 fn test_struct_unused_fields() {
-    let input = "    struct Point {
+    let input = "    struct A {
         x: int,
         y: int,
         z: int,
     }
-    let p: Point = Point{ y: 0, x: 2 };
+    let x = A{ y: 0, x: 2 };
+";
+
+    type_check(input, &[]);
+}
+
+#[test]
+#[should_panic = "Field 'y' specified more than once"]
+fn test_struct_repeated_fields_expr() {
+    let input = "    struct A {
+        x: int,
+        y: int,
+    }
+    let x = A{ y: 0, x: 2, y: 1 };
+";
+
+    type_check(input, &[]);
+}
+
+#[test]
+#[should_panic = "Field 'x' is already declared"]
```

Comment:
Why is this a different error message than above?

Reply:
Both errors are for the case of repeated fields, but "specified more than once" is for the instantiation and "is already declared" is for the declaration. It's the way rust handles them.